pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -188,6 +190,57 @@ def distributor(self, distributor_id):
         """
         return self._distributors_by_id.get(distributor_id)
 
+    def search_content(self, name=None, arch=None, filename=None, stream=None):
+        """Search this repository for content matching the given criteria.
+
+        Args:
+            name (str)
+                Name of the desired unit.
+            arch (str)
+                Architecture of the desired unit.
+            filename (str)
+                File name of the desired unit.
+            stream (str)
+                Stream of the desired modulemd or modulemd_defaults unit.
+
+        Returns:
+            Future[list[:class:`~pubtools.pulplib.Task`]]
+                A future which is resolved when search succeeds.
+
+                The future contains a list of zero or more tasks triggered and awaited
+                during the search operation.
+
+        Raises:
+            DetachedException
+                If this instance is not attached to a Pulp client.
+
+        .. versionadded:: 2.4.0
+        """
+        if not self._client:
+            raise DetachedException()
+
+        criteria = (
+            Criteria.with_field(
+                "type_ids",
+                Matcher.in_(["rpm", "srpm", "modulemd", "modulemd_defaults"]),
```

Comment:
I think what you should do here is something like this:
In pubtools/pulplib/_impl/model/unit/__init__.py  add something like SUPPORTED_UNITS = (FileUnit, RpmUnit,....) and also make this constant public.
and here you sould basically accept type_ids in search and check if type_ids are subset of SUPPORTED_UNITS.content_type_id - or accept list of Unit* classes directly as type_ids attribute.
More dynamic way would be to check for content of the module and grep for all classes with base "Unit" and then accepts type_ids based on that.

Reply:
Good ideas for validating the content type id. I'll be sure it's included.

Reply:
"if type_id not in [cls.__name__ for cls in Unit.__subclasses__()]" should do it.